database/migrations/2022_09_18_143201_make_wui_settings_public.php

```diff
@@ -0,0 +1,223 @@
+	public const CONF_CATEGORY_FOOTER = 'Footer';
+	public const CONF_CATEGORY_CONF = 'config';
+	public const CONF_CATEGORY_LANDING = 'Mod Welcome';
+
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 *
+	 * @throws InvalidArgumentException
+	 */
+	public function up(): void
+	{
+		// The "owner" information is not only used on the landing page,
+		// but in the general footer of the gallery as well.
+		// Rename that to reflect this.
+
+		DB::table('configs')
+			->where('key', '=', 'landing_owner')
+			->update([
+				'key' => 'site_owner',
+				'cat' => self::CONF_CATEGORY_CONF,
+				'confidentiality' => 0,
+			]);
+
+		// The social media links are not only used on the landing page,
+		// but also in the general footer of the gallery as well, if enabled.
+		// Rename them to reflect this and put them into their own category
+		// to keep them together.
+
+		DB::table('configs')
+			->where('key', '=', 'landing_facebook')
+			->update([
+				'key' => 'sm_facebook_url',
+				'cat' => self::CONF_CATEGORY_SOCIAL_MEDIA,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'landing_flickr')
+			->update([
+				'key' => 'sm_flickr_url',
+				'cat' => self::CONF_CATEGORY_SOCIAL_MEDIA,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'landing_twitter')
+			->update([
+				'key' => 'sm_twitter_url',
+				'cat' => self::CONF_CATEGORY_SOCIAL_MEDIA,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'landing_instagram')
+			->update([
+				'key' => 'sm_instagram_url',
+				'cat' => self::CONF_CATEGORY_SOCIAL_MEDIA,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'landing_youtube')
+			->update([
+				'key' => 'sm_youtube_url',
+				'cat' => self::CONF_CATEGORY_SOCIAL_MEDIA,
+				'confidentiality' => 0,
+			]);
+
+		// Make footer settings public and group them together
+
+		DB::table('configs')
+			->where('key', '=', 'display_social_in_gallery')
+			->update([
+				'key' => 'footer_show_social_media',
+				'cat' => self::CONF_CATEGORY_FOOTER,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'site_copyright_enable')
+			->update([
+				'key' => 'footer_show_copyright',
+				'cat' => self::CONF_CATEGORY_FOOTER,
+				'confidentiality' => 0,
+			]);
+		DB::table('configs')
+			->where('key', '=', 'additional_footer_text')
+			->update([
+				'key' => 'footer_additional_text',
+				'cat' => self::CONF_CATEGORY_FOOTER,
+				'confidentiality' => 0,
+			]);
+
+		// Make copyright settings available to WUI
+
+		DB::table('configs')
+			->where('key', '=', 'site_copyright_begin')
+			->update(['confidentiality' => 0]);
+		DB::table('configs')
+			->where('key', '=', 'site_copyright_end')
+			->update(['confidentiality' => 0]);
+
+		// Remove NSFW text setting which is replaced by a localized text
+		DB::table('configs')
+			->where('key', '=', 'nsfw_warning_text')
+			->delete();
```

Comment:
I'm not sure about this one. I mean, it's nice that the _default_ is localized now; that's an improvement. But you are effectively taking away from the users the ability to customize that text to their liking. We have received at least one report from somebody who found the available space (200 characters?) too small, so clearly there is interest in customizing it.

I would suggest to keep a config option (perhaps renamed) but make it empty by default, in which case the localized text is used.

Reply:
You got me. I know there was another thing which I wanted to pull out of this PR as an individual PR and this one was it.

Actually, I had a totally different reason why I removed this config option. That the NSFW text is now localized is just a nice side effect. I am aware of the problem that this takes away a feature and why I wanted it to make an indepedent PR.

In order to avoid code injection attacks, I attempted to only use the JS property `textContent` and not `innerHTML`. Previously, this configuration option contained HTML code like `<h1>...</h1><p>...</p><p>...</p>`. I wanted to get rid of that. That is also why the locale file now contains three distinct entries: one with the text content for each of these HTML elements.

The current HTML code looks like this
```html
<!-- NSFW Warning -->
<div id="sensitive_warning" class="overlay-container"><h1></h1><p></p><p></p></div>
```
and already predefines the three child elements. The locale defines three items
```php
'NSFW_BANNER_HEADER' => 'Sensitive content',
'NSFW_BANNER_EXPL' => 'This album contains sensitive content which some people may find offensive or disturbing.',
'NSFW_BANNER_CONSENT_TAP' => 'Tap to consent.',
```
and the JS code looks like
```js
const nsfwBanner = document.querySelector("#sensitive_warning");
nsfwBanner.children.item(0).textContent = lychee.locale["NSFW_BANNER_HEADER"];
nsfwBanner.children.item(1).textContent = lychee.locale["NSFW_BANNER_EXPL"];
nsfwBanner.children.item(2).textContent = lychee.locale["NSFW_BANNER_CONSENT_TAP"];
```

If we keep the configuration option, then I assume it should only be a single configuration option as it has been before. (I do not assume that we would make three configuration options out of it.) I further assume this single configuration option should probably support HTML markup in order to provide the most flexibility. This could be a potential gateway for injection attacks although one can argue that the configuration option is managed by administrators and administrators should now what they are doing. But in theory, an administrator could put a link to a JS script file into that configuration option in good faith and then the JS script might be susceptible to attacks putting the whole Lychee application at risk.

Of course, alternatively, we could provide _three_ distinct configuration options (one for the header, one for the text, one for the action) which only allow to change the text, but which are otherwise restricted to the given structure.

Reply:
Fixed it in https://github.com/LycheeOrg/Lychee/pull/1522/commits/8d3b9123794840885d7ceb723fe200ae7d53cefd and https://github.com/LycheeOrg/Lychee-front/pull/323/commits/b6b162e13f9ce0212c278a32381e44bc060753c2 at the risk of a potential code injection.